Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,193 @@
+  divisors of `n` such that the sum of that subset is equal to `n`. -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s ⊆ properDivisors n, ∑ i ∈ s, i = n
+
+/-- `n : ℕ` is a _weird_ number if and only if it is both abundant and pseudoperfect. -/
+def Weird (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ s ⊆ properDivisors n, ∑ i ∈ s, i ≠ n := by
+  rw [Pseudoperfect, not_and_or]
+  simp only [not_lt, nonpos_iff_eq_zero, mem_powerset, not_exists, not_and, ne_eq]
+
+theorem deficient_one : Deficient 1 := zero_lt_one
+theorem deficient_two : Deficient 2 := one_lt_two
+theorem deficient_three : Deficient 3 := by norm_num [Deficient]
+
+theorem abundant_twelve : Abundant 12 := by
+  rw [Abundant, show properDivisors 12 = {1,2,3,4,6} by rfl]
+  norm_num
+
+set_option maxRecDepth 1730 in
+theorem weird_seventy : Weird 70 := by
+  rw [Weird, Abundant, not_pseudoperfect_iff_forall]
+  have h : properDivisors 70 = {1, 2, 5, 7, 10, 14, 35} := by rfl
+  constructor
+  · rw [h]
+    repeat norm_num
+  · rw [h]
+    right
+    intro s hs
+    have hs' := mem_powerset.mpr hs
+    fin_cases hs' <;> decide
+
+lemma deficient_iff_not_abundant_and_not_perfect (hn : n ≠ 0) :
+    Deficient n ↔ ¬ Abundant n ∧ ¬ Perfect n := by
+  dsimp only [Perfect, Abundant, Deficient]
+  omega
+
+lemma perfect_iff_not_abundant_and_not_deficient (hn : 0 ≠ n) :
+    Perfect n ↔ ¬ Abundant n ∧ ¬ Deficient n := by
+  dsimp only [Perfect, Abundant, Deficient]
+  omega
+
+lemma abundant_iff_not_perfect_and_not_deficient (hn : 0 ≠ n) :
+    Abundant n ↔ ¬ Perfect n ∧ ¬ Deficient n := by
+  dsimp only [Perfect, Abundant, Deficient]
+  omega
+
+/-- A positive natural number is either deficient, perfect, or abundant -/
+theorem deficient_or_perfect_or_abundant (hn : 0 ≠ n) :
+    Deficient n ∨ Abundant n ∨ Perfect n := by
+  dsimp only [Perfect, Abundant, Deficient]
+  omega
+
+theorem Perfect.pseudoperfect (h : Perfect n) : Pseudoperfect n :=
+  ⟨h.2, ⟨properDivisors n, ⟨fun ⦃_⦄ a ↦ a, h.1⟩⟩⟩
+
+theorem Prime.not_abundant (h : Prime n) : ¬ Abundant n :=
+  fun h1 ↦ (h.one_lt.trans h1).ne' (sum_properDivisors_eq_one_iff_prime.mpr h)
+
+theorem Prime.not_weird (h : Prime n) : ¬ Weird n := by
+  simp only [Nat.Weird, not_and_or]
+  left
+  exact h.not_abundant
+
+theorem Prime.not_pseudoperfect (h : Prime p) : ¬ Pseudoperfect p := by
+  simp_rw [not_pseudoperfect_iff_forall, ← mem_powerset,
+    show p.properDivisors.powerset = {∅, {1}} by rw [Prime.properDivisors h]; rfl]
+  refine Or.inr (fun s hs ↦ ?_)
+  fin_cases hs <;>
+  simp only [sum_empty, sum_singleton] <;>
+  linarith [Prime.one_lt h]
+
+theorem Prime.not_perfect (h : Prime p) : ¬ Perfect p := by
+  have h1 := Prime.not_pseudoperfect h
+  revert h1
+  exact not_imp_not.mpr (Perfect.pseudoperfect)
+
+/-- Any natural number power of a prime is deficient -/
+theorem Prime.deficient_pow  (h : Prime n) : Deficient (n ^ m) := by
```

Comment:
Please add a version in terms of `IsPrimePow`

Reply:
Do you want a new theorem or a prove it with `IsPrimePow`? Do you want it as `IsDeficientPow`?

Reply:
```
theorem Prime.deficient_pow' (h : IsPrimePow (n ^ m)) : Deficient (n ^ m) := by
  dsimp [IsPrimePow] at h
  obtain ⟨p, ⟨k, ⟨h1, ⟨h2, h3⟩⟩⟩⟩ := h
  sorry
```
Like this?

Reply:
Okay I think I know what you mean. This is the proof I wrote.
```
theorem Prime.deficient_pow' (h : IsPrimePow n) : Deficient n := by
  obtain ⟨p, ⟨k, ⟨h1, ⟨_, h3⟩⟩⟩⟩ := h
  rw [← h3]
  refine deficient_pow (Prime.nat_prime h1)
```